pymongoexplain/explainable_collection.py

```diff
@@ -0,0 +1,92 @@
+import pymongo
+from json import dumps
+
+
+class BaseCommand():
+    def __init__(self, dictionary):
+        self.dictionary = self.convert_to_camelcase(dictionary)
+
+    def convert_to_camelcase(self, d: dict) -> dict:
+        ret = dict()
+        for key in d.keys():
+            if "_" in key:
+                new_key = key.split("_")[0] + ''.join(
+                    [i.capitalize() for i in key.split("_")[1:]])
+                ret[new_key] = d[key]
+            else:
+                ret[key] = d[key]
+        return ret
+
+    def get_SON(self):
+        return self.dictionary
+
+
+class UpdateCommand(BaseCommand):
+    def __init__(self, collection: pymongo.collection, filter, update,
+                 kwargs):
+        self.dictionary = {"update": collection.name,
+                           "updates": [{"q": filter, "u": update}]}
+        for key, value in kwargs.items():
+            self.dictionary["updates"][0][key] = value
+        super(UpdateCommand, self).__init__(self.dictionary)
+
+class DistinctCommand(BaseCommand):
+    def __init__(self, key, collection: pymongo.collection, filter, session,
+                 kwargs):
+        self.dictionary = {"distinct": collection.name, "key": key, "query":
+            filter}
+        for key, value in kwargs.items():
+            self.dictionary[key] = value
+        super(UpdateCommand, self).__init__(self.dictionary)
+
+class AggregateCommand(BaseCommand):
+    def __init__(self, collection: pymongo.collection, pipeline, kwargs):
+        self.dictionary = {"aggregate": collection.name, "pipeline": pipeline,
+                           "query": filter}
+        for key, value in kwargs.items():
+            self.dictionary[key] = value
+        super(UpdateCommand, self).__init__(self.dictionary)
+
+class CountCommand(BaseCommand):
+    def __init__(self, collection: pymongo.collection, filter,
+                 kwargs):
+        self.dictionary = {"count": collection.name,
+                           "query": filter}
+        for key, value in kwargs.items():
+            self.dictionary[key] = value
+        super(UpdateCommand, self).__init__(self.dictionary)
+
+class ExplainCollection():
+    def __init__(self, collection):
+        self.collection = collection
+
+    def _explain_command(self, command):
+        print({"explain": command.get_SON(), "verbosity": "queryPlanner"})
+        return self.collection.database.command(
+            {"explain": command.get_SON(), "verbosity": "queryPlanner"})
+
+    def update_one(self, filter, update, **kwargs):
```

Comment:
What do you think about explicitly adding all the arguments that Collection.update_one supports? Like:
```
    def update_one(self, filter, update, upsert=False,
                   bypass_document_validation=False,
                   collation=None, array_filters=None, hint=None,
                   session=None):
```
Maybe we can keep **kwargs at the end for better future pymongo compatibility?

Reply:
I think I prefer to use **kwargs and then simply check for whether certain kwargs are set rather than explicitly defining them because it makes it easier to pass onto the command function, but I might be wrong

Reply:
Yeah I agree it's easier, but it makes the API harder for users since they won't have autocomplete for the support arguments.

Reply:
That makes sense. I will make the change